src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkPath.java

```diff
@@ -27,24 +30,27 @@
 import net.starlark.java.eval.EvalException;
 import net.starlark.java.eval.Printer;
 import net.starlark.java.eval.Sequence;
+import net.starlark.java.eval.Starlark;
 import net.starlark.java.eval.StarlarkValue;
 import net.starlark.java.eval.Tuple;
 
 /**
- * A Path object to be used into Starlark remote repository.
+ * A Path object to be used in repo rules and module extensions.
  *
  * <p>This path object enable non-hermetic operations from Starlark and should not be returned by
- * something other than a StarlarkRepositoryContext.
+ * something other than a StarlarkBaseExternalContext.
  */
 @Immutable
 @StarlarkBuiltin(
     name = "path",
     category = DocCategory.BUILTIN,
     doc = "A structure representing a file to be used inside a repository.")
 public final class StarlarkPath implements StarlarkValue {
+  private final StarlarkBaseExternalContext ctx;
```

Comment:
Can a `StarlarkPath` instance stay alive after the repository rule finishes its execution? That same thing is very possible with configured targets, and if it happens with repository rules, the `StarlarkBaseExternalContext` instance would be acessible after the execution of the repository rule, which would be bad. The workaround for this is probably to access `StarlarkBaseExternalContext`from a thread local variable, as undesirable as that is.

Reply:
i see where you're coming from but fortunately no, StarlarkPath objects are never held onto (why would you anyway, it's literally a wrapper around a Path).

Reply:
What's the systemic guarantee for that? I guess module extension / repository functions don't actually have a Starlark value they can return?

Reply:
they can return a thing, but said thing cannot legally be a StarlarkPath (nor the rctx object itself, for that matter).

Reply:
Ack, now that I read through the code, that's indeed the case. It's a somewhat flimsy way to enforce the "rctx doesn't escape" invariant, though.